### PR TITLE
Submitting HTML forms.

### LINKED ISSUE #199: Support HTML forms
For a form like:

```html
<form method="POST" id="my-form" action="/register">
<input type="email" name="username" />
</form>
```

We may want an API such as:

```typescript
const someResource = ketting.go('/form.html');`
const state = await someResource.get();

const newState = await state.action('my-form').submit({
  name: 'me@evertpot.com',
});
```

cc @awwright

### PR DESCRIPTION
**Pull request by evert**
Fixes #199

### CODE REVIEW

File: src/util/html.ts [hunk 1]
```diff
@@ -1,39 +1,111 @@
...
+  }
 
   return links;
 
 }
+
+function parseForm(contextUri: string, node: sax.Tag): HtmlForm[] {
+
+  const rels = node.attributes.REL || null;
+  const id = node.attributes.ID || null;
```
Comment (juhangsin) [inline src/util/html.ts:81]: just wondering, is ID for this capitalized?

Comment (juhangsin) [reply src/util/html.ts:81]: ah, nvm, everything else in attributes are capitalized..

Comment (evert) [reply src/util/html.ts:81]: Yea that's how the `sax` library encodes attribute and tag names

Comment (evert) [reply src/util/html.ts:81]: good eye though

**Pull request merged**